state/accountsDB.go

```diff
@@ -918,7 +918,8 @@ func (adb *AccountsDB) commit() ([]byte, error) {
 		return nil, err
 	}
 
-	adb.stateChangesCollector.Reset()
+	//TODO: discuss the workflow. If reset here, the outport driver won't be able to pick up the changes.
+	//adb.stateChangesCollector.Reset()
```

Comment:
I would move the reset in the processor (shardProcessor & metaProcessor) in the CommitBlock, after calling `sp.indexBlockIfNeeded(bodyHandler, headerHash, headerHandler, lastBlockHeader)` on shardProcessor, and  
after calling `mp.indexBlock(header, headerHash, body, lastMetaBlock, notarizedHeadersHashes, rewardsTxs)` on metaProcessor

Reply:
moved

Reply:
don't forget to remove also the commented code after we agree on the workflow.